enterprise/server/test/integration/remote_cache/remote_cache_test.go

```diff
@@ -296,6 +296,37 @@ func TestBuild_RemoteCache_ScoreCard(t *testing.T) {
 	}
 }
 
+func TestBuild_RemoteCache_RejectsInvalidAPIKeys(t *testing.T) {
+	app := buildbuddy_enterprise.Run(t, "--auth.enable_anonymous_usage=true")
+	ctx := context.Background()
+	ws := testbazel.MakeTempWorkspace(t, workspaceContents)
+
+	{
+		// Build 1: explicit empty API key
+		testbazel.Clean(ctx, t, ws)
+		iid := newUUID(t)
+		buildFlags := []string{
+			"//:hello.txt", "--invocation_id=" + iid,
+			"--remote_header=x-buildbuddy-api-key=",
```

Comment:
Should we add a test to make sure that non-explicitly set empty keys (i.e. omitted keys) don't fail?

Reply:
Good call, this case is tested above, but not intentionally. Done